packages/openid4vc/src/openid4vc-verifier/OpenId4VcVerifierModule.ts

```diff
@@ -0,0 +1,130 @@
+import type { OpenId4VcVerifierModuleConfigOptions } from './OpenId4VcVerifierModuleConfig'
+import type { OpenId4VcVerificationRequest } from './router'
+import type { AgentContext, DependencyManager, Module } from '@aries-framework/core'
+import type { NextFunction } from 'express'
+
+import { AgentConfig } from '@aries-framework/core'
+
+import { getAgentContextForActorId, getRequestContext, importExpress } from '../shared/router'
+
+import { OpenId4VcSiopVerifierService } from './OpenId4VcSiopVerifierService'
+import { OpenId4VcVerifierApi } from './OpenId4VcVerifierApi'
+import { OpenId4VcVerifierModuleConfig } from './OpenId4VcVerifierModuleConfig'
+import { OpenId4VcVerifierRepository } from './repository'
+import { configureAuthorizationEndpoint } from './router'
+
+/**
+ * @public
+ */
+export class OpenId4VcVerifierModule implements Module {
+  public readonly api = OpenId4VcVerifierApi
+  public readonly config: OpenId4VcVerifierModuleConfig
+
+  public constructor(options: OpenId4VcVerifierModuleConfigOptions) {
+    this.config = new OpenId4VcVerifierModuleConfig(options)
+  }
+
+  /**
+   * Registers the dependencies of the question answer module on the dependency manager.
+   */
+  public register(dependencyManager: DependencyManager) {
+    // Warn about experimental module
+    const logger = dependencyManager.resolve(AgentConfig).logger
+    logger.warn(
+      "The '@aries-framework/openid4vc' Verifier module is experimental and could have unexpected breaking changes. When using this module, make sure to use strict versions for all @aries-framework packages."
+    )
+
+    // Register config
+    dependencyManager.registerInstance(OpenId4VcVerifierModuleConfig, this.config)
+
+    // Api
+    dependencyManager.registerContextScoped(OpenId4VcVerifierApi)
+
+    // Services
+    dependencyManager.registerSingleton(OpenId4VcSiopVerifierService)
+
+    // Repository
+    dependencyManager.registerSingleton(OpenId4VcVerifierRepository)
+  }
+
+  public async initialize(rootAgentContext: AgentContext): Promise<void> {
+    this.configureRouter(rootAgentContext)
+  }
+
+  /**
+   * Registers the endpoints on the router passed to this module.
+   */
+  private configureRouter(rootAgentContext: AgentContext) {
```

Comment:
Have not checked it out in great detail, but will the express servers work fine if you are an issuer AND verifier?

Reply:
Yes, you just have to make sure it's exposed on a different path. See https://github.com/animo/openid4vc-playground/blob/b2b039bd316914aaedfda752b1a99e9326c5ba55/agent/src/server.ts#L28-L29